demos/system_reference-ST.rst

```diff
@@ -143,7 +144,9 @@ From the zephy_rpmsg_multi_services directory
 Install the Zephyr binary on the sdcard
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The Zephyr sample binary is available in the sub-folder of build directory stm32mp15-demo/zephy_rpmsg_multi_services/build/zephyr/rpmsg_multi_services.elf. It needs to be installed on the "rootfs" partition of the sdcard
+The Zephyr sample binary is available in the sub-folder of build directory
+stm32mp15-demo/zephy_rpmsg_multi_services/build/zephyr/rpmsg_multi_services.elf.
+It needs to be installed on the "rootfs" partition of the sdcard
 
 .. code-block:: console
```

Comment:
Other notes:
WRT: https://openamp.readthedocs.io/en/latest/openamp-system-reference/examples/zephyr/rpmsg_multi_services/README.html

1) For your build all the modules are built in so the insmod line fails for all.  You should add a note.
2) The rpmgs_utils are built into the Xilinx vendor build and the openamp-ci builds so you should probibly move the section about building and installing them to the ST demo page.
3) The rpmsg_util readme not longer describes how to cross compile it so you should add that to your st demo page also.  In the past I have just used the ubuntu gcc-arm-linux-gnueabihf package but 24.04 glibc may be too new for nanbield build.  It did work for ubuntu 22.04. (make CC=arm-linux-gnueabihf-gcc)
4) On the other builds the utils are in the PATH not in the root user dir.  So add the utils to /usr/bin or similar so we can use the same syntax for all cases

Reply:
I also notice that:
1) shutdown now does not really shutdown
2) poweroff takes a bit but does poweroff but there are some kernel oops'es (i2 related) on the shutdown path

Reply:
> Other notes: WRT: https://openamp.readthedocs.io/en/latest/openamp-system-reference/examples/zephyr/rpmsg_multi_services/README.html 
> 2. The rpmgs_utils are built into the Xilinx vendor build and the openamp-ci builds so you should probibly move the section about building and installing them to the ST demo page.
> 3. The rpmsg_util readme not longer describes how to cross compile it so you should add that to your st demo page also.  In the past I have just used the ubuntu gcc-arm-linux-gnueabihf package but 24.04 glibc may be too new for nanbield build.  It did work for ubuntu 22.04. (make CC=arm-linux-gnueabihf-gcc)

I prefer to keep it in  rpmgs_utils,  we should be able to built it in a standalone way.  the Rpmsg utils need probably few update to not consider that i can be build on diferent platfrom not only arm. 
I will sent a separate PR to propose update based on your comments.

Reply:
WRT to rpmsg_util: The build instructions should really be in the README of that dir.
If you keep it in the rpmsg_util demo page PLEASE add something like:
"many OS images already have the rpmsg utilities included, if yours does not please follow these instructions:"

I checked glibc versions in YP & Ubuntu:
22.04: 2.35
24.04: 2.39
YP nanbield: 2.38
YP scarthgap (5.0): 2.39
ARM toolchain 13.1.Rel1 (Oct 2023): 2.38

So using Ubuntu packaged arm gcc may not work on nanbield.  I know your wiki says tested on 22.04 but that is rather hidden.  It would probibly be best to suggest using the arm toolchain.

Reply:
> WRT to rpmsg_util: The build instructions should really be in the README of that dir. If you keep it in the rpmsg_util demo page PLEASE add something like: "many OS images already have the rpmsg utilities included, if yours does not please follow these instructions:"
I updated related readme in https://github.com/OpenAMP/openamp-system-reference/pull/42


> 
> I checked glibc versions in YP & Ubuntu: 22.04: 2.35 24.04: 2.39 YP nanbield: 2.38 YP scarthgap (5.0): 2.39 ARM toolchain 13.1.Rel1 (Oct 2023): 2.38
> 
> So using Ubuntu packaged arm gcc may not work on nanbield. I know your wiki says tested on 22.04 but that is rather hidden. It would probibly be best to suggest using the arm toolchain.
I will add a dedicated chapter

Reply:
chapter to rebuild rpmsg-utils added